messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+        this.converters = builder.converters;
+        this.claimDuration = builder.claimDuration;
+    }
+
+    /**
+     * Creates a new builder, capable of building a {@link JpaSequencedDeadLetterQueue} according to the provided
+     * configuration. Note that the {@link Builder#processingGroup(String)} is mandatory for the queue to work
+     * correctly.
+     *
+     * @param <M> The message type
+     * @return The builder
+     */
+    public static <M extends EventMessage<?>> Builder<M> builder() {
+        return new Builder<>();
+    }
+
+    @Override
+    public void enqueue(@Nonnull Object sequenceIdentifier, @Nonnull DeadLetter<? extends M> letter)
+            throws DeadLetterQueueOverflowException {
+        String sequence = toIdentifier(sequenceIdentifier);
+        if (isFull(sequence)) {
+            throw new DeadLetterQueueOverflowException(
+                    "No room left to enqueue [" + letter.message() + "] for identifier ["
+                            + sequence + "] since the queue is full."
+            );
+        }
+
+        if (logger.isDebugEnabled()) {
+            Optional<Cause> optionalCause = letter.cause();
+            if (optionalCause.isPresent()) {
+                logger.debug("Adding dead letter [{}] because [{}].", letter.message(), optionalCause.get());
+            } else {
+                logger.debug("Adding dead letter [{}] because the sequence identifier [{}] is already present.",
+                             letter.message(), sequence);
+            }
+        }
+
+        DeadLetterEntryMessage converter = converters
+                .stream()
+                .filter(c -> c.canConvert(letter.message()))
+                .findFirst()
+                .map(c -> c.toEntry(letter.message(), serializer))
+                .orElseThrow(() -> new IllegalArgumentException("No converter found"));
+
+        transactionManager.executeInTransaction(() -> {
+            Long maxIndexForQueueIdentifier = getNextIndexForSequence(sequence);
+            DeadLetterEntry deadLetter = new DeadLetterEntry(processingGroup,
+                                                             sequence,
+                                                             maxIndexForQueueIdentifier,
+                                                             converter,
+                                                             letter.enqueuedAt(),
+                                                             letter.lastTouched(),
+                                                             letter.cause().orElse(null),
+                                                             letter.diagnostics(),
+                                                             serializer);
+            entityManager().persist(deadLetter);
+        });
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void evict(DeadLetter<? extends M> letter) {
+        if (!(letter instanceof JpaDeadLetter)) {
```

Comment:
I would expect a test case in the `JpaSequencedDeadLetterQueueTest` class for this check.